airbyte-integrations/bases/debezium-v1-9-6/src/main/java/io/airbyte/integrations/debezium/internals/PostgresConverter.java

```diff
@@ -106,6 +138,65 @@ private void registerText(final RelationalColumn field, final ConverterRegistrat
     });
   }
 
+  private Object convertArray(Object x, RelationalColumn field) {
+    String fieldType = field.typeName().toUpperCase();
+    Object[] values = new Object[0];
+    try {
+      values = (Object[]) ((PgArray) x).getArray();
+    } catch (SQLException e) {
+      LOGGER.error("Failed to convert PgArray:" + e);
+    }
+    switch (fieldType) {
+      case "_MONEY":
+
+        final String nativeMoneyValue = ((PgArray) x).toString();
+        final String substringM = Objects.requireNonNull(nativeMoneyValue).substring(1, nativeMoneyValue.length() - 1);
+        final char currency = substringM.charAt(0);
+        final String regex = "\\" + currency;
+        List<String> myListM = new ArrayList<>(Arrays.asList(substringM.split(regex)));
+        return myListM.stream()
+                .map(val -> val.replaceAll("[^\\d.]", ""))
+                .filter(money -> !money.isEmpty())
+                .map(Double::valueOf)
+                .collect(Collectors.toList());
+      case "_NUMERIC":
+        return Arrays.stream(values).map(value -> value == null ? null : Double.valueOf(value.toString())).collect(Collectors.toList());
+      case "_TIME":
+        return Arrays.stream(values).map(value -> value == null ? null : convertToTime(value)).collect(Collectors.toList());
+      case "_DATE":
+        return Arrays.stream(values).map(value -> value == null ? null : convertToDate(value)).collect(Collectors.toList());
+      case "_TIMESTAMP":
+        return Arrays.stream(values).map(value -> value == null ? null : convertToTimestamp(value)).collect(Collectors.toList());
+      case "_TIMESTAMPTZ":
+        return Arrays.stream(values).map(value -> value == null ? null : convertToTimestampWithTimezone(value)).collect(Collectors.toList());
+      case "_TIMETZ":
```

Comment:
> Why cant we use the `convertToTimeWithTimezone` method just like we are doing for the rest of the types?

debezium handles timetz and timetz [] differently
pgarray returns an array of java.sql.Time and not java.time.OffsetTime while the timezone that uses the customer is lost, so for timetz [] I use a custom implementation to transfer data without losing the timezone

Reply:
> Let's add this logic to method convertToTimeWithTimezone in class DateTimeConverter and add a condition like

> if (time instanceof final java.sql.Time timetz) then do what you are doing

@subodh1810 the approach of converting to  java.sql.Time is not entirely correct, because in this case we lose information about the time zone. That's why I refused to use it.
In my implementation, just java.lang.String is used and we shouldn't add this implementation in DateTimeConverter.convertToTimeWithTimezone() as it only applies to Debezium and Postgres array processing